daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/ValueCoder.scala

```diff
@@ -145,17 +131,15 @@ object ValueCoder {
     * converts the value to the type usable by engine/interpreter.
     *
     * @param protoValue0 the value to be read
-    * @param decodeCid a function to decode stringified contract ids
     * @tparam Cid ContractId type
     * @return either error or [VersionedValue]
     */
-  def decodeVersionedValue[Cid](
-      decodeCid: DecodeCid[Cid],
-      protoValue0: proto.VersionedValue,
-  ): Either[DecodeError, VersionedValue[Cid]] =
+  def decodeVersionedValue(
+      protoValue0: proto.VersionedValue
+  ): Either[DecodeError, VersionedValue] =
```

Comment:
We don’t use it anywhere atm so I think my inclination is to drop it and add it back if we do want to use it. wdyt?

Reply:
I assumed canton used it. 
I think it could be useful for ledger implementer to have a version of the decoder that does not accepts V0. 
I will rather audit if this is actually useless before dropping the code, otherwise we will never added it back. 

If we drop it know, we will never added it back, because 
- on the one hand this is not useful for use (the language team) 
- on the other hand actual users (canton/kv) very rarely give us feedback on the API we provide. 

Additionally, we could use this code to ensure that contract key does not have Cid at the serialization level. 

It is not clear to me that all of that would be actually useful, but
- the current code can be the current code can be very easily adapted to add extra cid checks. 
- I am pretty confident the current code is correct, I would be much less if we add it back.

Reply:
added it back in 9b39254ebf07d3c0f49baa36b46beae176dafa46. I’ve added both the de and encoder. It seems weird to be non-symmetric here and if you want to prevent it on decoding you might also want a failsafe on encoding.

Reply:
Indeed I want for both. 
Thanks for taking time to revert this change, even if is not a deal breaker for me.